src/udev/udev-builtin-net_id.c

```diff
@@ -581,6 +581,9 @@ static int get_device_firmware_node_sun(sd_device *dev, uint32_t *ret) {
         if (r < 0)
                 return log_device_warning_errno(dev, r, "Failed to parse firmware_node/sun '%s', ignoring: %m", attr);
 
+        if (*ret == 0)
```

Comment:
Please update `ret` only when the function succeeds.
```c
uin32_t sun;
r = safe_atou32(attr, &sun);
if (r < 0)
        return log_device_debug_errno(...);
if (sun == 0)
        return log_device_debug_errno(...);

*ret = sun;
return 0;
```

Reply:
Done